vermouth/processors/merge_all_molecules.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2018 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .processor import Processor
+
+
+class MergeAllMolecules(Processor):
+    """
+    Merge all the molecules from a system.
+
+    The molecules are merged into the first molecule of the system. Nothing is
+    done if there are no molecules.
+    """
+    def run_system(self, system):
+        if not system.molecules:
+            return system
+
+        molecule = system.molecules[0]
+        for other in system.molecules[1:]:
+            molecule.merge_molecule(other)
```

Comment:
Can't shouldn't this reuse code from MergeChains?

Reply:
It could, but the added logic of listing the chains and generating the inputs for MergeChain, and the added API is almost as much of adding the full new processor. I went for the simpler user interface, but I could leverage MergeChain.